packages/replay/src/auth.ts

```diff
@@ -263,3 +264,128 @@ export async function maybeAuthenticateUser(options: Options = {}) {
     return false;
   }
 }
+
+function parseId(id: string) {
+  return Buffer.from(id, "base64").toString("utf-8").split(":")[1];
+}
+
+async function getAuthInfo(key: string) {
+  const resp = await query(
+    "AuthInfo",
+    `
+        query AuthInfo {
+          viewer {
+            user {
+              id
+            }
+          }
+          auth {
+            workspaces {
+              edges {
+                node {
+                  id
+                }
+              }
+            }
+          }
+        }
+      `,
+    undefined,
+    key
+  );
+
+  if (resp.errors) {
+    throw {
+      id: "graphql-error",
+      message: resp.errors
+        .map((e: any) => e.message)
+        .filter(Boolean)
+        .join(", "),
+    };
```

Comment:
I'd prefer to always throw actual errors - `.stack` is super useful when diagnosing problems. You could introduce a `class GraphQL extends Error {}` here if it doesn't exist in the codebase yet.